openedx/core/djangoapps/import_export/courses/views.py

```diff
@@ -0,0 +1,529 @@
+            )
+        except Exception as exc:
+            log.exception(
+                'There was an error exporting course %s',
+                courselike_key
+            )
+            return self._export_error_response(
+                {
+                    "context_course": courselike_module.url_name,
+                    "error": True,
+                    "error_message": str(exc),
+                    "unit": None
+                },
+                redirect_url=redirect_url
+            )
+        finally:
+            shutil.rmtree(root_dir / name)
+
+        # The course is all set; return the tar.gz
+        wrapper = FileWrapper(export_file)
+
+        response = HttpResponse(wrapper, content_type='application/x-tgz')
+        response['Content-Disposition'] = 'attachment; filename={}'.format(
+            os.path.basename(
+                export_file.name.encode('utf-8')
+            )
+        )
+        response['Content-Length'] = os.path.getsize(export_file.name)
+        return response
+
+    # pylint: disable=unused-argument
+    @ensure_valid_course_key
+    @renderer_classes_decorator((JSONRenderer,))
+    def post(self, request, course_key_string):
+        """
+        The restful handler for importing a course.
+
+        GET
+            json: return json import status
+        POST or PUT
+            json: import a course via the .tar.gz file specified inrequest.FILES
+        """
+        courselike_key = CourseKey.from_string(course_key_string)
+        library = isinstance(courselike_key, LibraryLocator)
+
+        if library:
+            root_name = LIBRARY_ROOT
+            context_name = "context_library"
+            courselike_module = modulestore().get_library(courselike_key)
+            import_func = import_library_from_xml
+        else:
+            root_name = COURSE_ROOT
+            context_name = "context_course"
+            courselike_module = modulestore().get_course(courselike_key)
+            import_func = import_course_from_xml
+
+        filename = request.FILES['course-data'].name
+        courselike_string = unicode(courselike_key) + filename
+        data_root = path(settings.GITHUB_REPO_ROOT)
+        subdir = base64.urlsafe_b64encode(repr(courselike_key))
+        course_dir = data_root / subdir
+
+        status_key = "{}|{}".format(
+            request.user.username,
+            courselike_string
+        )
+
+        # Do everything in a try-except block to make sure everything is
+        # properly cleaned up.
+        try:
+            # Cache the import progress
+            self._save_request_status(request, courselike_string, 0)
+            if not filename.endswith('.tar.gz'):
+                self._save_request_status(request, courselike_string, -1)
+                return JsonResponse(
+                    {
+                        'error_message': _(
+                            'We only support uploading a .tar.gz file.'
```

Comment:
Yes, it's shown to users.

Could you let me know of this error formatting is okay?: https://github.com/edx/edx-platform/pull/6190/files#diff-49938df912930fb450d16ae2b949a463R462